index.js

```diff
@@ -260,10 +260,12 @@ export default {
           }
 
           // FIXME: make this async
-          rawTx.nonce = web3http.eth.getTransactionCount(
-            fromEthAddress,
-            rawTx.fromFullShardKey,
-          );
+          if (!rawTx.nonce) {
```

Comment:
how about `if (rawTx.nonce == undefined)` which will cover both `undefined` and `null` cases but not `0`

Reply:
commit 4756aada574c520c29a0dce807514124a2809838